### ISSUE TITLE
Resources - Credits box

### ISSUE THREAD

**Issue by jamesd-uk** [OpenSprites/OpenSprites-next#29]
There should be a credits box on every resource page. The user has the option to select a license from a drop down menu. The options are:
- [Public Domain](https://creativecommons.org/share-your-work/public-domain/cc0/)
- [Attribution 2.0 Generic](https://creativecommons.org/licenses/by/2.0/)
- [Attribution-ShareAlike 2.0 Generic](https://creativecommons.org/licenses/by-sa/2.0/)
- [Attribution-NonCommercial 2.0 Generic](https://creativecommons.org/licenses/by-nc/2.0/)
- [Attribution-NoDerivs 2.0 Generic](https://creativecommons.org/licenses/by-nd/2.0/)
- [Attribution-NonCommercial-ShareAlike 2.0 Generic](https://creativecommons.org/licenses/by-nc-sa/2.0/)
- [Attribution-NonCommercial-NoDerivs 2.0 Generic](https://creativecommons.org/licenses/by-nc-nd/2.0/)

There should also be a text field to specify the author.

**Comment by bates64** [comment]
Bit fancy for a kid-oriented site; perhaps default to _whatever the scratch one is_? And default the author to "you" (translated to "username" when not logged in) or something

**Comment by jamesd-uk** [comment]
Yeah, I know it is a bit too complicated, but we can't modify the license of content upon upload.

**Comment by jamesd-uk** [comment]
Perhaps have a simplified license description in the drop down and in the credits box, and then a question mark next to it which gives the actual license names and links to them? For example:
- "Attribution-NonCommercial 2.0 Generic" is shown as "I must give credit to the original creator and not use this resource to make a profit".
- "Attribution-NonCommercial-NoDerivs 2.0 Generic" is shown as "I must give credit to the original creator, not use this content to make a profit, and not share any changes I've made to this resource".

**Comment by plastpappa** [comment]
I'd say we go with @nanalan s approach, so when you're uploading something you can specify a license and if you don't a popup explaining why you might need to specify a certain one pops up and you can check "Don't show me again" and choose either "Choose a license" or "Use the default license" (or something like it).  
It encourages a more advanced license use in a friendly way while still allowing an option for those not wanting to bother with it.

**Comment by plastpappa** [comment]
We don't really need to allow for license-changing after upload if we do that. Your other suggestion is great as well, especially in combination with what I suggested.

**Comment by jamesd-uk** [comment]
Yeah, I would agree with what you've said above. I've mostly been thinking about content uploaded by users which has not been created by them, which is where most of my concerns about legality are.

**Comment by bates64** [comment]
We can always mark stuff with the default as needing to be checked by a mod (google image search, anyone?) to look for licensing stuff - and if it's copyrighted delete the file and give the user an alert?

**Comment by plastpappa** [comment]
Anything with no credits should be marked too. We should have pop-ups like the one I suggested for license defaulting but for explaining the consequences of stealing others work.

**Comment by jamesd-uk** [comment]
I think that is a good measure to put in place. An extra couple of minutes checking that stuff could save heaving to deal with angry copyright owners.

**Comment by Arinerron** [comment]
+1. And maybe even a page to briefly and simply explain the licenses so young people could understand it.

Ex:

**Custom License** states:
- You must give credit to the original author
- You may copy and redistribute
- You may use the resource commercially 
  ...etc...

**Comment by bates64** [comment]
@Arinerron pretty sure kvack mentioned that :package:

**Comment by Arinerron** [comment]
@nanalan I wrote that message before the second response to this issue, but realized I hadn't clicked send :P
